docs/manuals/gui/extension/index.md

```diff
@@ -0,0 +1,212 @@
+Other methods can be overridden if necessary, which will be discussed later in the manual.
+
+## Declaring Elements
+
+As we have seen, custom visual element descriptors are associated with the name of
+the element as it appears on pages in the `(ElementLibrary.)get_elements()^` method.
+
+The [`Element` constructor](Element.__init__()^) needs a description of all the properties
+that this element holds, as well as how this element is rendered.
+
+```py
+Element("<default_property_name>",
+       {
+           "<property_1_name>": ElementProperty(<property_1_type>, ...),
+           "<property_2_name>": ElementProperty(<property_2_type>, ...),
+           ...
+        },
+        <rendering_arguments>)
+```
+
+An element property descriptor (handled by the `ElementProperty^` class) must
+indicate its type (one of the `PropertyType^` values) and potentially a default
+value.
+
+All elements must indicate what their default property name is. That is used in Markdown
+pages as the first fragment of the `<|...|>` construct or the value located in the
+text part of an element tag in HTML pages.
+
+Of course, *default_property_name* must be one of the keys of the properties dictionary
+provided to the [`Element` constructor](Element.__init__()^).
+
+## Rendering Elements
+
+The Taipy GUI Extension package provides two different ways to implement the rendering
+of the element and its interactions. Both approaches deliver HTML fragments that are
+inserted into the page when it is requested.
+
+- Static elements.<br/>
+  A static visual element cannot interact with the underlying application.
+  It can be used just like any other element in a page, but its properties are not
+  bound to the application variable: if the variable value is modified, it does not
+  impact the representation of the element on the page.<br/>
+  Static elements are implemented by creating a string that holds the XHTML text (that is,
+  HTML that respects the XML syntax, where all tags must be closed) that is inserted in
+  the page displayed by the browser.<br/>
+  Please go to the [Static Elements Example](extension_static_element.md) page for
+  a complete description of how to implement your own static custom elements.
+
+- Dynamic elements.<br/>
+  Dynamic elements provide the binding functionality of Taipy GUI: if a property value
+  depends on an application variable (and if the property type is *dynamic*), then the
+  page automatically updates when the variable value changes.<br/>
+  In Taipy GUI, dynamic visual elements are implemented using the
+  [React](https://reactjs.org/) JavaScript library and the
+  [TypeScript](https://www.typescriptlang.org/) programming language (that builds on
+  JavaScript).
+
+## Prerequisites
+
+To create and use custom visual elements, you need to install the following:
+
+- Taipy GUI 2.0 or higher (included in Taipy and Taipy Enterprise).
+- Python 3.8 or higher.
+- NodeJS (and NPM) if you need to create dynamic custom visual elements.
```

Comment:
minimal version of NodeJs ? 18 ?

Reply:
18, ok.
I was trying to avoid saying it in case it changes.